lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -384,30 +384,73 @@ class DeclAndTypePrinter::Implementation
     os << "@end\n";
   }
 
+  void visitEnumDeclCxx(EnumDecl *ED) {
+    // FIXME: Print enum's availability
+    ClangValueTypePrinter printer(os, owningPrinter.prologueOS,
+                                  owningPrinter.typeMapping,
+                                  owningPrinter.interopContext);
+    printer.printValueTypeDecl(ED, /*bodyPrinter=*/[&]() {
+      ClangSyntaxPrinter syntaxPrinter(os);
+      llvm::SmallVector<const EnumElementDecl *> elements;
+
+      os << "  enum class cases {\n";
+      for (const auto *caseDecl : ED->getAllCases()) {
+        for (const auto *element : caseDecl->getElements()) {
+          os << "    ";
+          syntaxPrinter.printIdentifier(element->getNameStr());
+          elements.push_back(element);
+          os << ",\n";
+        }
+      }
+      os << "  };\n"; // enum class cases' closing bracket
+
+      if (elements.size() == 0) {
+        os << "\n";
+        return;
+      }
+
+      // Rearrange cases so that they match their tag values
+      llvm::stable_sort(elements, [&](const auto *e1, const auto *e2) {
```

Comment:
How did you determine that this is the correct ordering? Is the compiler doing this kind of sort anywhere else? We might want to reuse this if thins is already being done elsewhere

Reply:
I got it from the memory layout doc, but yeah it's better to reuse existing functionality. I'll ask this in the office hour meeting.